### ISSUE TITLE
How to custom train a model to detect rooftops?

### ISSUE THREAD

**Issue by Vijay-P1999** [azavea/raster-vision#1384]
## ❓ Questions and Help

Hello,
I need to detect rooftops and roads in tiff images. I tried using the pre-trained models [model zoo](https://docs.rastervision.io/en/0.13/examples.html#model-zoo).
But none of them predicted correctly. So i guess, i've to train a custom model. Can you provide some insights on how to train and label the tif images?

##  Already tried
1. [CVAT](https://cvat.org/)
2. [labelme](https://github.com/wkentaro/labelme)
Both couldn't handle 9gb tif image.

**Comment by theoway** [comment]
You can use QGIS for labelling.

**Comment by AdeelH** [comment]
You might also want to consider GroundWork: https://groundwork.azavea.com/

**Comment by Vijay-P1999** [comment]
> You might also want to consider GroundWork: https://groundwork.azavea.com/

Hey Adeel, Thank you. I tried GroundWork and it will be very much useful for my case. 
But as i mentioned in my issue, my tif image is ~9gb in size. I couln't upload it as a single image. I tiled it and upload the tiles. 

My question is, should i label all the tiles and then train using those tiles only or should i label the whole single image?

And how many images do you recommend for training from scratch? P.S for semantic segmentation of buildings and roads

**Comment by AdeelH** [comment]
>My question is, should i label all the tiles and then train using those tiles only or should i label the whole single image?

Have you uploaded each tile as a separate GroundWork project in a single campaign?

**Comment by Vijay-P1999** [comment]
No, i just uploaded single tile in a campaign. Any suggestions?

**Comment by AdeelH** [comment]
I would do the following:
- break your full image up into tiles and upload each as a project in a single campaign
- label them
- export your campaign as a STAC catalog
  - if you have fully labeled everything, then just export using the GroundWork GUI
  - if only partially labeled, you will need to do an export via GroundWork's API
    - so that GW exports an AOI layer that specifies which parts of the image have been labeled
    - see the code cells under the "From GroundWork to new training data" heading in this notebook: https://github.com/azavea/foss4g-groundwork-hitl-infra/blob/ogc-workshop/upload-script/HITL_upload.ipynb
- read data from that catalog in your Raster Vision `get_config()` function
  - example: https://github.com/azavea/foss4g-groundwork-hitl-infra/blob/ogc-workshop/upload-script/active_learning.py
- make each tile a separate Raster Vision scene
- split the scenes into train/val/test however you want

I can provide more details if you get stuck on any of these.
